pilot/pkg/model/validation.go

```diff
@@ -2030,7 +2030,17 @@ func validateCORSPolicy(policy *networking.CorsPolicy) (errs error) {
 		return
 	}
 
-	// TODO: additional validation for AllowOrigin?
+	for _, host := range policy.AllowOrigin {
+		if host != "*" {
+			if strings.HasPrefix(host, "https://") {
+				host = host[8:]
+			}
+			if strings.HasPrefix(host, "http://") {
+				host = host[7:]
+			}
+			errs = appendErrors(errs, ValidateFQDN(host))
```

Comment:
Isn't an optional `:port` allowed? ValidateFQDN will fail for `example.com:80`, won't it?

Reply:
Good point.  I found https://tools.ietf.org/html/rfc6454#section-4 which says an origin may include a port.  I found a few places (e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Origin ) that say only the server name is allowed, but I think the IETF document is authoritative.

Reply:
You are correct. Updated the PR.